packages/calcite-components/src/components/text-area/text-area.tsx

```diff
@@ -410,7 +415,7 @@ export class TextArea
       this.localizedCharacterLengthObj = this.getLocalizedCharacterLength();
       return (
         <span class={CSS.characterLimit}>
-          <span class={{ [CSS.characterOverLimit]: this.value?.length > this.maxLength }}>
+          <span class={{ [CSS.characterOverLimit]: this.isCharacterLimitExceeded() }}>
```

Comment:
To support context with NVDA and FF, add a unique `id`, which can be referenced in the `aria-describedby` in the `textarea`, similar to [`combobox-item`](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/combobox-item/combobox-item.tsx#L228), where a guid is created.